axlearn/common/attention.py

```diff
@@ -1758,6 +1734,7 @@ def _forward_for_mode(
             query: A Tensor or TensorSpec of shape [batch, target_length, target_dim].
             key:   An optional Tensor of shape [batch, source_length, source_dim].
             value: An optional Tensor of shape [batch, source_length, source_dim].
+            paddings: An optional Tensor of shape [batch, target_dim] for query paddings.
```

Comment:
nit: add more comments on the semantics of paddings?

Reply:
it's changed to live_step_len and has long docstring.